examples/addVectors.cxx

```diff
@@ -0,0 +1,192 @@
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-689114
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For additional details, please also read RAJA/LICENSE.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the disclaimer below.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the disclaimer (as noted below) in the
+//   documentation and/or other materials provided with the distribution.
+//
+// * Neither the name of the LLNS/LLNL nor the names of its contributors may
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY,
+// LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <iostream>
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/util/defines.hpp"
+
+#define CUDA_BLOCK_SIZE 256
+
+void checkSolution(int *C, int in_N);
+
+/*
+  Memory management is left to the developer
+*/
+int * allocate(RAJA::Index_type size) {
+  int * ptr;
+#if defined(RAJA_ENABLE_CUDA)
+  cudaMallocManaged((void**)&ptr, sizeof(int)*size,cudaMemAttachGlobal);
+#else
+  ptr = new int[size];
+#endif
+  return ptr;
+}
+
+void deallocate(int* &ptr) {
+  if (ptr) {
+#if defined(RAJA_ENABLE_CUDA)
+    cudaFree(ptr);
+#else
+    delete[] ptr;
+#endif
+    ptr = nullptr;
+  }
+}
+
+/*  
+  Example 1: Adding Two Vectors
+
+  ----[Details]---------------------
+  Adds two vectors of length N
+
+  -----[RAJA Concepts]---------------
+  1. Introduces the forall loop and basic RAJA policies
+
+  RAJA::forall<RAJA::exec_policy>(RAJA::range_policy,[=](index i)) {
+
+         //body
+
+  });
+  
+  [=] Pass by copy
+  [&] Pass by reference
+  RAJA::exec_policy  - specifies how the traversal occurs
+  RAJA::range_policy - provides a list in which the index may iterate on
```

Comment:
RAJA::range_policy should more properly be Iterable, or perhaps segment, these days.  Keeping policies and iterables separate may help keep the concepts clear.